File: App/src/Adaptive.ReactiveTrader.Client/UI/Blotter/TradeViewModel.cs
using System;
using System.Globalization;
using Adaptive.ReactiveTrader.Client.Domain.Models;
using Adaptive.ReactiveTrader.Shared.UI;
using PropertyChanged;

namespace Adaptive.ReactiveTrader.Client.UI.Blotter
{
    [ImplementPropertyChanged]
    public class TradeViewModel : ViewModelBase, ITradeViewModel
    {
        public TradeViewModel(ITrade trade)
        {
            TradeId = trade.TradeId.ToString(CultureInfo.InvariantCulture);
            CurrencyPair = trade.CurrencyPair.Substring(0, 3) + " / " + trade.CurrencyPair.Substring(3, 3);
            Direction = trade.Direction == Domain.Models.Direction.Buy ? "Buy" : "Sell";
            Notional = trade.Notional.ToString("N0", CultureInfo.InvariantCulture) + " " + trade.DealtCurrency;
            SpotRate = trade.SpotRate;
            TradeDate = trade.TradeDate;
            TradeStatus = trade.TradeStatus == Domain.Models.TradeStatus.Done ? "Done" : "REJECTED";
            TraderName = trade.TraderName;
            ValueDate = trade.ValueDate;
            DealtCurrency = trade.DealtCurrency;
        }

        public decimal SpotRate { get; set; }
        public string Notional { get; set; }
        public string Direction { get; set; }
        public string CurrencyPair { get; set; }
        public string TradeId { get; private set; }
        public DateTime TradeDate { get; private set; }
        public string TradeStatus { get; private set; }
        public string TraderName { get; private set; }
        public DateTime ValueDate { get; private set; }
        public string DealtCurrency { get; private set; }
    }
}

File: src/Adaptive.ReactiveTrader.Client.JS/UI/Blotter/TradeViewModel.ts
 class TradeViewModel implements ITradeViewModel {
     spotRate: number;
     notional: string;
     direction: string;
     currencyPair: string;
     tradeId: string;
     tradeDate: string;
     tradeStatus: string;
     traderName: string;
     valueDate: string;
     dealtCurrency: string;

     constructor(trade: ITrade) {
         this.spotRate = trade.spotRate;
         this.notional = trade.notional + " " + trade.dealtCurrency;
         this.direction = trade.direction == Direction.Buy ? "Buy" : "Sell";
         this.currencyPair = trade.currencyPair.substring(0, 3) + " / " + trade.currencyPair.substring(3, 6);
         this.tradeId = trade.tradeId.toFixed(0);
         this.tradeDate = trade.tradeDate.toString();
         this.tradeStatus = trade.tradeStatus == TradeStatus.Done ? "Done" : "REJECTED";
         this.traderName = trade.traderName;
         this.valueDate = trade.valueDate.toString();
         this.dealtCurrency = trade.dealtCurrency;
     }
 }